lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -708,6 +708,10 @@ private function getPreAlterTableAlterPrimaryKeySQL(TableDiff $diff, Index $inde
 
         // Dropping primary keys requires to unset autoincrement attribute on the particular column first.
         foreach ($index->getColumns() as $columnName) {
+            if(!$diff->fromTable->hasColumn($columnName)) {
```

Comment:
sorry to nitpick but we use the following code style `if (! $diff` can you please adjust that? afterwards I'll merge.